pypit/fluxspec.py

```diff
@@ -0,0 +1,431 @@
+        # Step
+        self.steps.append(inspect.stack()[0][3])
+        # Return
+        return self.std
+
+    def master(self, row_fitstbl, clobber=False):
+        """
+        Load or generate+save the MasterFrame sensitivity function
+        The solution is applied to the list of science spectra loaded (if any)
+
+        Parameters
+        ----------
+        row_fitstbl : Row
+          Row of the fitstbl corresponding to the Standard star
+          Used to parse RA, DEC, AIRMASS, EXPTIME
+        clobber : bool, optional
+
+        Returns
+        -------
+        self.sensfunc
+
+        """
+        self.sensfunc, _, _ = self.load_master_frame()
+        # Sensitivity Function
+        if (self.sensfunc is None) or clobber:
+            if self.std_specobjs is None:
+                msgs.warn("You need to load in the Standard spectra first!")
+                return None
+
+            # Find the star automatically?
+            _ = self.find_standard()
+
+            # Kludge a header
+            if self.std_header is None:
+                self.std_header={}
+                for key in ['ra','dec','airmass','exptime']:
+                    self.std_header[key.upper()] = row_fitstbl[key]
+
+            # Sensitivity
+            _ = self.generate_sensfunc()
+
+            # Save to master
+            self.save_master()
+
+        # Apply to science
+        if len(self.sci_specobjs) > 0:
+            self.flux_science()
+        # Return
+        return self.sensfunc
+
+    def save_master(self, outfile=None):
+        """
+        Over-load the save_master() method in MasterFrame to write a YAML file
+
+        Parameters
+        ----------
+        outfile : str, optional
+          Use this input instead of the 'proper' (or unattainable) MasterFrame name
+
+        Returns
+        -------
+
+        """
+        # Step
+        self.steps.append(inspect.stack()[0][3])
+        # Allow one to over-ride output name
+        if outfile is None:
+            outfile = self.ms_name
+        # Add steps
+        self.sensfunc['steps'] = self.steps
+        # yamlify
+        ysens = arutils.yamlify(self.sensfunc)
+        with open(outfile, 'w') as yamlf:
+            yamlf.write(yaml.dump(ysens))
+        #
+        msgs.info("Wrote sensfunc to MasterFrame: {:s}".format(outfile))
+
+    def show_sensfunc(self):
+        """
+        Plot the sensitivity function
+        """
+        if self.sensfunc is None:
+            msgs.warn("You need to generate the sensfunc first!")
+            return None
+        # Generate from model
+        wave = np.linspace(self.sensfunc['wave_min'], self.sensfunc['wave_max'], 1000)
+        mag_func = arutils.func_val(self.sensfunc['c'], wave, self.sensfunc['func'])
+        sens = 10.0**(0.4*mag_func)
+        # Plot
+        debugger.plot1d(wave, sens, xlbl='Wavelength', ylbl='Sensitivity Function')
+
+
+    def write_science(self, outfile):
```

Comment:
Add a method to write sensfunc to disk via a .yaml file that does not rely on running the save_master racket.

Reply:
Already exists as part of save_master(), e.g.

FxSpec.save_master(outfile='name_this_yaml_file_what_you_want.yaml')